packages/itwin/grouping-mapping-widget/tsconfig.json

```diff
@@ -4,8 +4,14 @@
     "skipLibCheck": true,
     "jsx": "react",
     "preserveSymlinks": true,
+    "resolveJsonModule": true,

```

Comment:
I think you would change the `QueryBuilder.testdata.json` file into `QueryBuilder.testdata.ts`. That way you will not need to add the `resolveJsonModule` rule and it will be easier to see what properties are required for a specific testcase. Also, this will allow comments for test cases.

Reply:
Done

Reply:
First time I've seen `.testdata.ts`. Is this typical?

Reply:
I don't know if you can see it that often, but the idea is to have a single test method for a lot of tests where only the inputs and expectations change. The behavior is always the same - add some properties, remove some properties, build the query, check the result. Having all the test data in `.test.ts` file could make it harder to read the test itself. At first we thought that `.json` file would be a good candidate for test data, but it has some drawbacks:
- cannot have comments
- types are not enforced, so there is high chance you'll spend hours debugging the test just to find a typo

`.testdata.ts` can have both - comments and types. And as a bonus - `resolveJsonModule` is not required.